extension/json/json_common.cpp

```diff
@@ -31,24 +32,65 @@ string ThrowPathError(const char *ptr, const char *end, const bool binder) {
 	}
 }
 
-static inline idx_t ReadString(const char *ptr, const char *const end, const bool escaped) {
+struct JSONKeyReadResult {
+public:
+	static inline JSONKeyReadResult Empty() {
+		return {idx_t(0), string()};
+	}
+
+	static inline JSONKeyReadResult WildCard() {
+		return {1, "*"};
+	}
+
+	inline bool IsValid() {
+		return chars_read != 0;
+	}
+
+	inline bool IsWildCard() {
+		return key == "*";
+	}
+
+public:
+	idx_t chars_read;
+	string key;
```

Comment:
Should this be a `string` or can this be a `string_t`? Since this is called for every value it might be worth avoiding the extra allocation/copy.

Reply:
It needs to be a `string` because we have to construct the unescaped string (we can't just point to the original), so we need an allocation. I think this can be optimized at some point, where instead of going through the JSON path every time, we construct an internal representation of the path that can be reused. The optimization can be done in a later PR since this can be considered a bugfix